### ISSUE TITLE
Feature request: Control Hyperdeck metadata (shot, take, etc.)

### ISSUE THREAD

**Issue by PhotoJoseph** [bitfocus/companion-module-bmd-hyperdeck#77]
Whenever I import hyperdeck media into Resolve, there's always some metadata already in place, and of course its all wrong. There's camera name, scene shot take, etc. Probably a lot more. If there was a way to control that… ooh that'd be lovely!!

**Comment by sphlabs** [comment]
I'm working on this now, and it occurs to me that there is a good case for exposing some of the project info on the config page of a Hyperdeck connection. Project name, Director, Camera & Camera Operator could all be configured on a per-machine basis, as that info doesn't change very often.

There can also be actions to change this info, so one wouldn't need to go back to the config page to make changes, but I can see a lot of scenarios where it would be nice to set, say, the camera ID once, and always have that saved to all clips on that deck.

I don't think this approach makes sense for the clip info (take, reel, scene, etc.) but I'm open to suggestions.

**Comment by LikeThisReview** [comment]
Stephen, Putting project meta data on the Edit Connection page means one cannot access them in variables?? When a clip is selected in “playback mode” we need to see the project values from the clip. When we are about to record we are able to set the project meta date (slate for next). See also user manual Hyperdeck Extreme (a bit more info then in the manual for  “regular” models). 
Also the Reel info in the connection is not really whats needed (format the SD Card for camera A and the new format name should change from A001 in to A002 (or A003). Could be done under the button if it is a variable)

Regards, Nico Wasmus

**Comment by LikeThisReview** [comment]
Stephen, 
If possible, please create variables for all meta data. Information on Edit Connection page is not preferred (as in current setup we can’t access this info like Reel). Want to display that info on buttons as well.

A content creator wants to change project name for each youtube video for example?! But OK, if there is an action for this.

It looks like Black magic no longer refers to project name as such but product name (I guess because they use project name now for the cloud).

Indeed Clip info needs to be all variables, if possible a set for current clip and fir next clip (slate for next)

If camera id = A and Reel is 8 the SD Card is with volume name A008 and file suffix is also A008.

**Comment by PhotoJoseph** [comment]
> I'm working on this now, and it occurs to me that there is a good case for exposing some of the project info on the config page of a Hyperdeck connection. Project name, Director, Camera & Camera Operator could all be configured on a per-machine basis, as that info doesn't change very often.

Agreed; it'd be great to get this info in there, and right this doesn't change often (if ever in some setups)

> There can also be actions to change this info, so one wouldn't need to go back to the config page to make changes, but I can see a lot of scenarios where it would be nice to set, say, the camera ID once, and always have that saved to all clips on that deck.

Right; I have six decks and while I can route any camera to any deck, it's almost always the same. So yes asigning the camera number to the deck would be extremely useful. 

> I don't think this approach makes sense for the clip info (take, reel, scene, etc.) but I'm open to suggestions.

Reel, Scene and Take are the primary fields used. It's interesting that there are many more fields you can use, but if you look at Blackmagic's own field recorders, the Video Assists, the only shot metadata you can adjust on there is Reel, Scene and Take. There's also a "good take last clip" button that's pretty awesome. Making a button on the streamdeck that tells all hyperdecks to mark the last shot as a good take would be amazing. 

In case it helps, as a comparison to what BMD does with its own hardware, here are photos of the metadata pages on the Video Assist recorders:

![IMG_0419](https://github.com/bitfocus/companion-module-bmd-hyperdeck/assets/78106443/dbcbcf13-34fa-4ec1-b45b-a1af80cb8409)
![IMG_0420](https://github.com/bitfocus/companion-module-bmd-hyperdeck/assets/78106443/38d6f09f-6446-4a38-a3ac-f08bb02f964d)
![IMG_0421](https://github.com/bitfocus/companion-module-bmd-hyperdeck/assets/78106443/e8a43cee-cfee-4a42-b488-ba3c967f77be)

**Comment by LikeThisReview** [comment]
All,
The images from PhotoJoseph does not show all information; if you tap Scene
and Take you are able to select shot type  and take type plus set it to
auto take number. See user manual ursa camera. iPone app does support A
take, but not shot type. The iPhone app does not allow for changing Lens
data as it is able to determine some of that data automatically.

I will post my setup for meta data pages in Companion in the Companion User
Group today / tomorrow.
Regards,
Nico Wasmus

**Comment by PhotoJoseph** [comment]
Thanks Nico -- I was showing the fields that are populated by the device, not _how_ they are populated. But yes, if you wanted to completely recreate the Video Assist experience, here are those pages. 

![IMG_0422](https://github.com/bitfocus/companion-module-bmd-hyperdeck/assets/78106443/80a113e5-cbe8-41be-ad26-66e230a50c97)
![IMG_0424](https://github.com/bitfocus/companion-module-bmd-hyperdeck/assets/78106443/9d022c73-977d-4e1c-9c41-2bccfb84d7f0)

**Comment by LikeThisReview** [comment]
👍

Op ma 11 mrt 2024 om 13:07 schreef PhotoJoseph ***@***.***>

**Comment by sphlabs** [comment]
Thanks for all the comments and feedback.

In terms of project data on the config page, I see this as being optional and useful for people who do things like ISO recording, where these settings will rarely change for a deck. Think of them as startup settings as, by setting in config they get pushed to the deck on connection without needing to push any buttons. [aside: in my testing the Shuttle does maintain project settings between power-cycles, but I do wonder if that is the case for all models]

My plan is that all project settings are still exposed as variables, and settable through actions, so there wouldn't be any loss of functionality in that way. The question would be what should happen to the config settings if an action changes those values… I'm inclined to leave the config fields unchanged and emphasise the 'startup values' state of them. The downside is that the config screen could potentially show values that are not currently saved in the deck (although a quick Save would fix that).

As for the clip (and lens) data, that would all be kept to actions/variables/feedback. The VideoAssist interface (I have one of those as well) is very similar to the Hyperdeck Extreme, which is why I believe all of these settings are present in other decks, as the rest only allow you to set the Camera through the interface.

**Comment by PhotoJoseph** [comment]
All sounds reasonable to me. Perhaps difficult to truly evaluate until trying it, so I'm of course more than happy to beta test for you and provide feedback.

**Comment by LikeThisReview** [comment]
Stephen,
Great. I'm willing to beta-test as well (on Windows 11).

Also the HyperDeck Studio HD Plus (ethernet protocol version: 1.13)
remembers the
values (test data):


slate clips

223 slate info:

slate for: *hpdR_0008.mp4*

reel: 1

scene id: 1

shot type: none

take: 60

take scenario: SER

good take: none

environment: interior

day night: day

take auto inc: true

preview: enable: true

200 ok


slate clips

223 slate info:

slate for: *Next Clip*

reel: 1

scene id: 1

shot type: none

take: 62

take scenario: SER

good take: none

environment: interior

day night: day

take auto inc: true

slate project

223 slate info:

project name: tcp intro

camera: B

director: none

camera operator: none

slate lens

223 slate info:

lens type: none

iris: none

focal length: none

distance: none

filter: none

I'm able to set all these values through the Generic TCP module and using
Triggers (with notify slate) to update my persistent customer variables so
they are in sync with the device and across multiple machines (MAc,
mini-PC).

Regards,
Nico

Op ma 11 mrt 2024 om 13:40 schreef Stephen Harrison <
***@***.***>:

**Comment by PhotoJoseph** [comment]
Hi @sphlabs have you found the time to make any progress on this? We're all excited to see 😊

cheers
-Joseph

**Comment by PhotoJoseph** [comment]
@sphlabs can we collectively offer some financial compensation to you to move this forward? I'd really love to see this added, and I'm sure many others here would as well. It's definitely worth $$ to me. I can't speak for @LikeThisReview but I'd be happy to pony up some cash 😊

**Comment by LikeThisReview** [comment]
PhotoJoseph, Happy to contribute in activities like design and testing for
free, but I’m saving my money for a 2 M/E Constellation switcher so I can
test my atem profiles for all BM switcher models. I’m a retired IT Project
Manager and configuring Companion is my hobby (without revenues).
Regards, Nico Wasmus @LikeThisReview / @StreamdeckControlFreak

**Comment by sphlabs** [comment]
Financial compensation is always nice, and that would certain help me push it up the priority list. The other factor is time, and the last few months have just been crazy busy with ‘real’ work. Things do start to slow down a bit next month though.Most of the work is already done, just needs a bit of polish (and updating the underlying Hyperdeck-connection library) before it can go out for beta testing.Joseph, I think you already have my email address, so feel free to reach out and I’m sure we can work something out.On 20 Jun 2024, at 12:24, LikeThisReview ***@***.***> wrote:﻿
PhotoJoseph, Happy to contribute in activities like design and testing for

free, but I’m saving my money for a 2 M/E Constellation switcher so I can

test my atem profiles for all BM switcher models. I’m a retired IT Project

Manager and configuring Companion is my hobby (without revenues).

Regards, Nico Wasmus @LikeThisReview / @StreamdeckControlFreak
